activerecord/lib/active_record/railtie.rb

```diff
@@ -29,10 +29,22 @@ class Railtie < Rails::Railtie # :nodoc:
       'ActiveRecord::RecordNotSaved'   => :unprocessable_entity
     )
 
-
     config.active_record.use_schema_cache_dump = true
     config.active_record.maintain_test_schema = true
 
+    def (config.active_record).destroy_all_in_batches=(value)
+      if value == false
```

Comment:
I'm not sure if we should really issue the warning in 5.0; or where to issue it in the code base, so folks updating to 5.0 can see it. Now the people will only see the warning if they know about the option and explicitly set it to `false`.